node/src/blockchain/blockchain_bridge.rs

```diff
@@ -358,27 +358,9 @@ impl BlockchainBridge {
                     debug!(self.logger, "Writing max_block_count({})", max_block_count);
                     self.persistent_config
                         .set_max_block_count(Some(max_block_count))
-                        .map_or_else(
-                            |_| {
-                                warning!(self.logger, "{} update max_block_count to {}. Scheduling next scan with that limit.", e, max_block_count);
-                                Err(format!("{} updated max_block_count to {}. Scheduling next scan with that limit.", e, max_block_count))
-                            },
-                            |e| {
-                                warning!(self.logger, "Writing max_block_count failed: {:?}", e);
-                                Err(format!("Writing max_block_count failed: {:?}", e))
-                            },
-                        )
-                } else {
-                    warning!(
-                        self.logger,
-                        "Attempted to retrieve received payments but failed: {:?}",
-                        e
-                    );
-                    Err(format!(
-                        "Attempted to retrieve received payments but failed: {:?}",
-                        e
-                    ))
+                        .unwrap_or_else(|_| panic!("Writing max_block_count failed: {:?}", e));
```

Comment:
As you can see, this changed from a treatable error to a panic, however, I cannot see any modification following this also in the test tree below. It implies that this panic is not tested. It surprises me because there must have been some tests covering this which should be panicking now. So maybe that test was truly missing even before. That's not an excuse for the present.

You will probably need to write a test which seems to be missing. Unless you find any sew this hole in our test coverage.